centaur/src/main/scala/centaur/api/CentaurCromwellClient.scala

```diff
@@ -61,58 +66,39 @@ object CentaurCromwellClient {
+  def metadata(workflow: SubmittedWorkflow): IO[WorkflowMetadata] = metadata(workflow.id)
 
-  def metadata(id: WorkflowId): Try[WorkflowMetadata] = {
+  def metadata(id: WorkflowId): IO[WorkflowMetadata] = {
     sendReceiveFutureCompletion(() => cromwellClient.metadata(id)) map { m =>
       WorkflowMetadata.fromMetadataJson(m.value).toOption.get
     }
   }
 
-  lazy val backends: Try[CromwellBackends] = sendReceiveFutureCompletion(() => cromwellClient.backends)
-
-  // The total time waiting for a Future, including network hiccups, must be longer than the time for a restart.
-  private val awaitTotalTimeout: FiniteDuration = CromwellManager.timeout * 2
-  private val awaitSleep: FiniteDuration = 5.seconds
-  private val awaitMaxAttempts: Long = awaitTotalTimeout.toSeconds / awaitSleep.toSeconds
-
-  /**
-    * Ensure that the Future completes within the specified timeout. If it does not, or if the Future fails,
-    * will return a Failure, otherwise a Success
-    */
-  def awaitFutureCompletion[T](x: () => Future[T], timeout: FiniteDuration, attempt: Int = 1): Try[T] = {
-    def sleepAndRetry(throwable: Throwable): Try[T] = {
-      if (LogFailures) Console.err.println(s"Failed to execute request to Cromwell, retrying: ${throwable.getMessage}")
-      Thread.sleep(awaitSleep.toMillis)
-      awaitFutureCompletion(x, timeout, attempt + 1)
-    }
+  lazy val backends: Try[CromwellBackends] = Try(Await.result(cromwellClient.backends, CromwellManager.timeout * 2))
 
-    // We can't recover the future itself with a "recoverWith retry pattern" because it'll timeout anyway from the Await.result
-    // We want to keep timing out to catch cases where Cromwell becomes unresponsive
-    Try(Await.result(x(), timeout)) recoverWith {
-      case throwable if attempt >= awaitMaxAttempts => Failure(throwable)
-      case throwable@(_: TimeoutException |
-                      _: StreamTcpException |
-                      _: IOException |
-                      _: UnsupportedContentTypeException) =>
-        sleepAndRetry(throwable)
-      case unsuccessful: UnsuccessfulRequestException if unsuccessful.httpResponse.status == StatusCodes.NotFound =>
-        sleepAndRetry(unsuccessful)
-      case unexpected: RuntimeException
-        if unexpected.getMessage.contains("The http server closed the connection unexpectedly") =>
-        // https://github.com/akka/akka-http/issues/768
-        sleepAndRetry(unexpected)
-      case exception@BufferOverflowException(message) if message.contains("Please retry the request later.") =>
-        // http://doc.akka.io/docs/akka-http/current/scala/http/client-side/pool-overflow.html
-        sleepAndRetry(exception)
-    }
+  def retryedRequest[T](x: () => Future[T], timeout: FiniteDuration): IO[T] = {
+    // If Cromwell is known not to be ready, delay the request to avoid requests bound to fail 
+    val ioDelay = if (!CromwellManager.isReady) IO.sleep(10.seconds) else IO.unit
```

Comment:
10 seconds is always enough?

Reply:
Probably not but it'll still be retried anyway below, it's just to avoid spamming retried requests